site/profile/manifests/metrics.pp

```diff
@@ -1,18 +1,125 @@
-class profile::metrics::exporter {
+class profile::metrics::node_exporter {
   include prometheus::node_exporter
   consul::service { 'node-exporter':
     port  => 9100,
-    tags  => ['monitor'],
+    tags  => ['node-exporter'],
     token => lookup('profile::consul::acl_api_token')
   }
 }
 
+class profile::metrics::slurm_job_exporter {
+  consul::service { 'slurm-job-exporter':
+    port  => 9798,
+    tags  => ['slurm-job-exporter'],
+    token => lookup('profile::consul::acl_api_token')
+  }
+
+  exec { 'pip install prometheus-client':
+    command => '/usr/bin/pip3.6 install prometheus-client',
+    unless  => '/usr/bin/pip3.6 freeze | /usr/bin/grep prometheus-client',
+    before  => Service['slurm-job-exporter'],
+  }
+
+  package { 'python3-psutil': }
+  -> package { 'slurm-job-exporter':
```

Comment:
Is the dependency (`->`) necessary here?

Reply:
`python3-psutil` was already a requirement from the RPM, removed from puppet since yum will install it.

Reply:
Misspoke, since slurm-job-exporter is installed without a repo, it will not install the required dependency of python3-psutil from the specfile, so it needs to be installed manually before slurm-job-exporter